corehq/apps/geospatial/static/geospatial/js/models.js

```diff
@@ -663,19 +663,28 @@ hqDefine('geospatial/js/models', [
         };
 
         self.clearSelectedPolygonFilter = function () {
+            if (!clearDisbursementBeforeProceeding()) {
+                return;
+            }
+
+            self.selectedSavedPolygonId('');
+            self.clearActivePolygon();
+            updateSelectedSavedPolygonParam();
+        };
+
+        function clearDisbursementBeforeProceeding() {
+            let proceedFurther = true;
```

Comment:
Nit: You could default this to `false` and then the `else` conditional below can be entirely removed.

Reply:
We want to proceed further in the case where there are no disbursement layers. So we could set it to either true/false by default however we will have one else in either case. 
(Although line 684 is not required for the current case)

Reply:
Ah good point, I didn't think of that path. In that case it's fine to leave this as is.